tools/Dashboard/DevHome.Dashboard/Extensions/ServiceExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.Dashboard.ViewModels;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace DevHome.Settings.Extensions;
+public static class ServiceExtensions
+{
+    public static IServiceCollection AddDashboard(this IServiceCollection services, HostBuilderContext context)
+    {
+        // View-models
+        services.AddSingleton<DashboardViewModel>();

```

Comment:
Today this behaves as Transient instead of Singleton. Looking at `DashboardViewModel`, with a singleton object lifetime, the value of `ShowDashboardBanner` will never change after object initialization while the app is running.

Reply:
This view model is a good candidate for a singleton object. Though in the future we have to ensure that showing the banner (from the settings page) would refresh the state of `ShowDashboardbanner`. One way to do that in the future is by binding the `ShouldShowDashboardBanner()` [relay command] to the View's `Loaded` event so that we refresh the value of `ShowDashboardBanner` every time we load the page while the view model object remains a singleton.